webapp/channels/src/components/sidebar/sidebar_channel/sidebar_channel_menu/sidebar_channel_menu.tsx

```diff
@@ -305,7 +438,26 @@ const SidebarChannelMenu = (props: Props) => {
             {favoriteUnfavoriteMenuItem}
             {muteUnmuteChannelMenuItem}
             <Menu.Separator/>
-            <ChannelMoveToSubmenu channel={props.channel}/>
+            {categories && (
+                <Menu.SubMenu
+                    id={Menu.createSubMenuId('moveTo', props.channel.id)}
+                    labels={
+                        <FormattedMessage
+                            id='sidebar_left.sidebar_channel_menu.moveTo'
+                            defaultMessage='Move to...'
+                        />
+                    }
+                    leadingElement={<FolderMoveOutlineIcon size={18}/>}
+                    trailingElements={<ChevronRightIcon size={16}/>}
+                    menuId={`moveTo-${props.channel.id}-menu`}
+                    menuAriaLabel={formatMessage({
+                        id: 'sidebar_left.sidebar_channel_menu.moveTo.dropdownAriaLabel',
+                        defaultMessage: 'Move to submenu',
+                    })}
+                >
+                    {getMoveToCategorySubmenuItems(categories, currentCategory)}
+                </Menu.SubMenu>
```

Comment:
🗒️ channelMoveToSubMenu is brought into this parent component

Reply:
I think that moving this submenu back into the component is actually a problem since it probably means that each channel in the sidebar is doing a lot of work to calculate which options should be available here despite the menu not being open.

I was originally going to suggest that we could just require custom submenu components to pass the injected props to its children, but it's a bit of a pain, and I think TS doesn't like doing that.

Maybe a better idea would be to use `Context` to pass `forceCloseMenu` down instead of directly injecting it as a prop? The `Menu` can wrap its children in a `Provider` and the `Menu.Item` can pick that click handler up with `useContext` without requiring the `Menu.Item` to be a direct child  of the `Menu`. Also, the `SubMenu` can add its own `Provider` that passes its own `forceCloseMenu` that  closes both the main menu and submenu. I think that'd also let us more deeply nest submenus as well.

Sorry, I forgot that `cloneElement` causes issues since it requires things to be direct children of one another which causes a bunch of problems. `Context` should 
allow us to have components like `ChannelMoveToMenu` and add intermediate components like `ChannelPermissionGate` too

Reply:
changed the implementation to context